docs/qdmi/qiskit_backend.md

```diff
@@ -0,0 +1,176 @@
+---
+file_format: mystnb
+kernelspec:
+  name: python3
+mystnb:
+  number_source_lines: true
+---
+
+# Qiskit Backend Integration
+
+The {py:mod}`mqt.core.qdmi.qiskit` module provides a Qiskit {py:class}`~qiskit.providers.BackendV2`-compatible interface to QDMI devices via FoMaC.
+
+## Installation
+
+Install MQT Core with Qiskit support:
+
+::::{tab-set}
+:sync-group: installer
+
+:::{tab-item} {code}`uv` _(recommended)_
+:sync: uv
+
+```console
+$ uv pip install "mqt-core[qiskit]"
+```
+
+:::
+
+:::{tab-item} {code}`pip`
+:sync: pip
+
+```console
+(.venv) $ python -m pip install "mqt-core[qiskit]"
+```
+
+:::
+::::
+
+## Quickstart
+
+```{code-cell} ipython3
+from mqt.core.qdmi.qiskit import QiskitBackend
+from qiskit import QuantumCircuit
+
+# Create a backend (uses the FoMaC device at index 0, i.e., MQT NA Default QDMI Device)
+backend = QiskitBackend(device_index=0)
```

Comment:
I started by looking at the documentation here. So I haven't looked at the actual code yet. I think that this is not the best API for managing multiple devices.
The notion of a `Provider` that manages multiple backends seems like a much better fit here. See https://quantum.cloud.ibm.com/docs/en/api/qiskit/providers#provider for an example of that.
I also don't think that we should assume that the NA device is at index 0 here. This seems overly restrictive.